scripts/models/TownHall.js

```diff
@@ -128,12 +128,43 @@
     });
   };
 
-  TownHall.lookupReps = function (zip) {
-    var representativePromise = $.ajax({
-      url: 'https://congress.api.sunlightfoundation.com/legislators/locate?zip=' + zip,
-      dataType: 'jsonp'
+  TownHall.lookupReps = function (districts) {
+    return new Promise(function (resolve, reject) {
+      var state = districts[0].split('-')[0];
+      districts = districts.map(function(district) {
+        return parseInt(district.split('-')[1]);
+      });
+      districts.push(null);
+
+      // Sunlight allows you to filter by districts...but if you do then senators don't show up, so we pull them all and look
+      // for those with matching districts or no district at all.
+      $.ajax({
+        url: 'https://congress.api.sunlightfoundation.com/legislators?per_page=all&state=' + state,
```

Comment:
yeah, sure. It would be good to know the pros and cons of both. Like I said before, google makes me slightly nervous since sometimes they drop their passion projects, but it seems like this has been going strong, and they do seem to have the most up to date zip information, which is why I'm using that to update our own database. Pro-publica also looks like it's updated some things since last I looked, but I dont know if it's just an update to the docs and not the api itself.